src/main/daml/Daml/Finance/Claims/Util/Builders.daml

```diff
@@ -191,22 +191,21 @@ createEuropeanCashClaim dateToTime ownerReceives strike spot cashInstrumentCid e
                                     else Lte (Observe spot, Const strike)
     claimAmount = when (TimeGte expiryDate) $
                   cond autoExerciseCondition exercised notExercised
-    claim = if ownerReceives then claimAmount else give claimAmount
+    claim = if isLong then claimAmount else give claimAmount
   in
     prepareAndTagClaims dateToTime [claim] "Option payment"
 
 -- | Create the claim for a physically settled European option.
 createEuropeanPhysicalClaim : (Date -> Time) -> Bool -> Decimal -> Deliverable -> Deliverable ->
   Date -> Bool -> TaggedClaim
-createEuropeanPhysicalClaim dateToTime ownerReceives strike underlying ccy
+createEuropeanPhysicalClaim dateToTime isLong strike underlying ccy
   expiryDate isCall =
   let
     exerciseClaim =
-      if isCall then
+      if isCall && isLong || not isCall && not isLong then
```

Comment:
maybe worth adding brackets just to make it easier to read (or maybe use `isCall /= isLong` as an xor and swap line 206 with 208)